eox_nelp/signals/tasks.py

```diff
@@ -10,76 +10,98 @@
 from django.contrib.auth.models import User
 from eox_core.edxapp_wrapper.courseware import get_courseware_courses
 from eox_core.edxapp_wrapper.enrollments import get_enrollment
+from eox_core.edxapp_wrapper.grades import get_course_grade_factory
 from opaque_keys.edx.keys import CourseKey
 from social_django.models import UserSocialAuth
 
 from eox_nelp.api_clients.futurex import FuturexApiClient
 from eox_nelp.edxapp_wrapper.course_overviews import CourseOverview
 
 courses = get_courseware_courses()
+CourseGradeFactory = get_course_grade_factory()
 logger = logging.getLogger(__name__)
 
 
+def _user_has_passing_grade(user, course_id):
+    """Determines if a user has passed a course based on the grading policies.
+
+    Args:
+        user<User>: Instace of Django User model.
+        course_id<str>: Unique course identifier.
+    Returns:
+        course_grade.passed<bool>: True if the user has passed the course, otherwise False
+    """
+    course_grade = CourseGradeFactory().read(user, course_key=CourseKey.from_string(course_id))
+
+    return course_grade.passed
+
+
 @shared_task
-def dispatch_futurex_progress(course_id, user_id):
-    """Dispatch the progress of a user in course to Futurex.
+def dispatch_futurex_progress(course_id, user_id, is_complete=None):
+    """Dispatch the course progress of a user to Futurex platform.
 
     Args:
-        course_id (str): user_id to dispatch the progress.
-        user_id (str): course_id to analyze.
+        course_id (str): Unique course identifier.
+        user_id (str): User identifier.
+        is_complete (bool): Determines is that hast complete the course
     """
     user = User.objects.get(id=user_id)
-    completion_summary = get_completion_summary(user, course_id)
-    progress_enrollment_data = generate_progress_enrollment_data(user, course_id, completion_summary)
+    user_has_passing_grade = is_complete if is_complete is not None else _user_has_passing_grade(user, course_id)
+
+    progress_enrollment_data = _generate_progress_enrollment_data(
+        user=user,
+        course_id=course_id,
+        user_has_passing_grade=user_has_passing_grade,
+    )
 
-    post_futurex_progress(progress_enrollment_data)
+    _post_futurex_progress(progress_enrollment_data)
 
 
-def post_futurex_progress(data):
+def _post_futurex_progress(data):
     """Make post for enrollment http request to FuturexApiClient
 
     Args:
         data (dict): dict to send to futurex enrollment-progress path.
     """
     api_client = FuturexApiClient()
-    enrollment_progress_path = "enrollment-progress?_format=json"
-    response = api_client.make_post(enrollment_progress_path, data)
+    response = api_client.enrollment_progress(data)
+
     logger.info(
         "send_futurex_progress --- The data %s  was sent to the url %s. The futurex response was: %s",
         data,
-        f"{api_client.base_url}/{enrollment_progress_path}",
```

Comment:
@andrey-canon why you remove this friendly line?

Reply:
beacuse enrollment_progress_path doesn't exist

Reply:
what do u think of keeping in the log the base domain? only api_client.base_url